aes/src/autodetect.rs

```diff
@@ -0,0 +1,201 @@
+//! Autodetection support for hardware accelerated AES backends with fallback
+//! to the fixsliced "soft" implementation.
+
+use crate::{Block, ParBlocks};
+use cipher::{
+    consts::{U16, U24, U32, U8},
+    generic_array::GenericArray,
+    BlockCipher, BlockDecrypt, BlockEncrypt, NewBlockCipher,
+};
+
+cpuid_bool::new!(aes_cpuid, "aes");
```

Comment:
We either need to add `ssse3`  to this list or use a separate `aes_ssse3_cpuid` module as you did before. Also it may be worth to a comment that SSE2 is implied by AES-NI, so we don't need to check for it separately. Or we could simply add `sse2` to the list to be completely thorough. :)

Reply:
Per my comments above, as far as I can tell every CPU with AES-NI has *both* SSE2 and SSSE3.

AES-NI was introduced in the [Westmere](https://en.wikipedia.org/wiki/Westmere_(microarchitecture)) architecture, which also has SSSE3. Unless there's some strange AMD/other CPU that has AES-NI but not SSSE3, I don't think it will be a problem.

Reply:
Per [reference](https://doc.rust-lang.org/reference/attributes/codegen.html#x86-or-x86_64) we can omit `sse2` check if we have verified that `aes` or `ssse3` is enabled, but it does not indicate any implicit dependency between `aes` and `ssse3`. So even though in practice there is probably no such CPU, for our code to be correct we have to check both those features.

Reply:
Hmm ok, sorry just merged but I can follow up with this.